qemu/tests/disk_extension.py

```diff
@@ -0,0 +1,168 @@
+      it is in io-error status,Otherwise it is in running status.
+
+    :param test: QEMU test object.
+    :param params: Dictionary with the test parameters.
+    :param env: Dictionary with test environment.
+    """
+
+    def cleanup_test_env(dirname, loop_device_name):
+        logging.debug("clean test %s %s" % (dirname, loop_device_name))
+        cmd = "if losetup -l {0};then losetup -d {0};fi".format(
+            loop_device_name)
+        process.system_output(cmd, shell=True)
+        cmd = "umount -l {0};rm -rf {0}".format(dirname)
+        process.system_output(cmd, shell=True)
+
+    def prepare_tmpfs_folder(dirname):
+        logging.debug("Ready to create tmpfs folder on %s" % dirname)
+        cmd = "umount -l {0};rm -rf {0};mkdir -p {0};".format(dirname)
+        process.system_output(cmd, ignore_status=True, shell=True)
+        cmd = "mount -t tmpfs -o rw,nosuid,nodev,seclabel tmpfs {}".format(
+            dirname)
+        process.system_output(cmd, shell=True)
+
+    def create_image_on_loop_device(backend_img, device_img):
+        if backend_img.image_format != "raw":
+            test.error(
+                "Only support attaching loop device on raw format image.")
+
+        backend_img.create(backend_img.params)
+        backend_filename = backend_img.image_filename
+        loop_device_name = device_img.image_filename
+        logging.debug("Ready to create loop device image {} on {}".format(
+            loop_device_name, backend_filename))
+        cmd = "losetup -d {}".format(loop_device_name)
+        process.system_output(cmd, ignore_status=True, shell=True)
+        cmd = "losetup {0} {1} && chmod 666 {0}".format(loop_device_name,
+                                                        backend_filename)
+        process.system_output(cmd, shell=True)
+        device_img.create(device_img.params)
+        logging.debug("Finish creating loop device image {} on {}".format(
+            loop_device_name, backend_filename))
+
+    def update_loop_device_backend_size(backend_img, device_img, size):
+        cmd = "qemu-img resize -f raw {} {}".format(backend_img.image_filename,
+                                                    size)
+        process.system_output(cmd, shell=True)
+        # update loop device
+        cmd = "losetup -c {0}".format(device_img.image_filename)
+        process.system_output(cmd, shell=True)
+
+    error_context.context("Get required parameter from the config file",
+                          logging.info)
+    current_size = int(params["begin_size"][0:-1])
+    max_size = int(params["max_size"][0:-1])
+    increment_size = int(params["increment_size"][0:-1])
+    size_unit = params["increment_size"][-1]
+    guest_cmd = params["guest_cmd"]
+
+    loop_device_backend_img_tag = params["loop_device_backend_img_tag"]
+    loop_device_img_tag = params["loop_device_img_tag"]
+
+    loop_device = params["loop_device"]
```

Comment:
I think this `loop_device = params["loop_device"]` is redundance.
The following call `cleanup_test_env(tmpfs_folder, loop_device)`, you could `cleanup_test_env(tmpfs_folder, params["loop_device"])`.